tests/syscall/mount-overlayfs.bats

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env bats
+
+#
+# Verify trapping & emulation on "mount" and "unmount2" syscalls for overlayfs mounts
+#
+
+
+load ../helpers/run
+load ../helpers/syscall
+load ../helpers/docker
+load ../helpers/sysbox-health
+
+ovs_path="/var/lib/docker/overlay2"
+
+function teardown() {
+  sysbox_log_check
+}
+
+# Verify basic overlayfs mount instruction.
+@test "mount overlayfs basic" {
+
+  local lower_path="$ovs_path"/root/lower
```

Comment:
I would also add files and subdir under lower and upper:

lower/l1
lower/ldir/l2
upper/u1
upper/udir/u2

And then verify that the merged dir has the correct hierarchy: 

l1
ldir/l2
u1
udir/u2

That way we know that ovfs it's doing its thing.

Reply:
This is a very basic test to simply verify if mount completes. The other testcases go much deeper into what you are suggesting.

Reply:
I didn't see any other test that does what I suggested above (but maybe I missed it).

Reply:
Oh i see what you mean: add files, folders, AND files-within-folders. Sure, will add this extra level to the 'addition' testcase.